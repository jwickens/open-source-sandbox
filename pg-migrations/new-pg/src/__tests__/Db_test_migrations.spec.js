/**
 * @flow
 */

const { Db } = require('../')
const path = require('path')

const OPTIONS = {
  schemaVersion: 0.1,
  schemaBase: 'db_test_migration',
  sqlDirectory: path.join(__dirname, 'sql')
}

describe('using pg to test migrations', () => {
  test('before migration the table and columns dont match', async () => {
    let before: Db, after: Db, beforeDesc: string, afterDesc: string
    try {
      before = new Db(OPTIONS)
      after = new Db(OPTIONS)
      await before.sync()
      await after.sync()
      await after.migrateTo(1)
      beforeDesc = await before.describe()
      afterDesc = await after.describe()
    } finally {
      before && await before.close()
      after && await after.close()
    }
    expect(beforeDesc).not.toEqual(afterDesc)
    expect(beforeDesc).toMatchSnapshot()
    expect(afterDesc).toMatchSnapshot()
  })
  test('after migration the table and columns match', async () => {
    let before: Db, after: Db, beforeDesc: string, afterDesc: string
    try {
      before = new Db(OPTIONS)
      after = new Db(OPTIONS)
      await before.sync()
      await after.sync()
      await after.migrateTo(1)
      await before.migrateTo(1)
      beforeDesc = await before.describe()
      afterDesc = await after.describe()
    } finally {
      before && await before.close()
      after && await after.close()
    }
    expect(beforeDesc).toEqual(afterDesc)
    expect(beforeDesc).toMatchSnapshot()
  }, 10000)
})
