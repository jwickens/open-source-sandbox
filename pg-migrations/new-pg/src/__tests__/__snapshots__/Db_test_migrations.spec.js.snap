// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using pg to test migrations after migration the table and columns match 1`] = `
Object {
  "functions": Array [
    "js_time(timestamp with time zone)",
  ],
  "tables": Object {
    "test": Array [
      Object {
        "arrayDataType": null,
        "camelName": "id",
        "columnName": "id",
        "dataType": "integer",
      },
      Object {
        "arrayDataType": null,
        "camelName": "testName",
        "columnName": "test_name",
        "dataType": "text",
      },
      Object {
        "arrayDataType": "text",
        "camelName": "tags",
        "columnName": "tags",
        "dataType": "ARRAY",
      },
      Object {
        "arrayDataType": null,
        "camelName": "message",
        "columnName": "message",
        "dataType": "text",
      },
    ],
    "version": Array [
      Object {
        "arrayDataType": null,
        "camelName": "version",
        "columnName": "version",
        "dataType": "numeric",
      },
      Object {
        "arrayDataType": null,
        "camelName": "deployStart",
        "columnName": "deploy_start",
        "dataType": "timestamp with time zone",
      },
      Object {
        "arrayDataType": null,
        "camelName": "deployedTs",
        "columnName": "deployed_ts",
        "dataType": "timestamp with time zone",
      },
      Object {
        "arrayDataType": null,
        "camelName": "clientCount",
        "columnName": "client_count",
        "dataType": "integer",
      },
      Object {
        "arrayDataType": null,
        "camelName": "payload",
        "columnName": "payload",
        "dataType": "jsonb",
      },
      Object {
        "arrayDataType": null,
        "camelName": "payloadVersion",
        "columnName": "payload_version",
        "dataType": "numeric",
      },
    ],
  },
}
`;

exports[`using pg to test migrations before migration the table and columns dont match 1`] = `
Object {
  "functions": Array [
    "js_time(timestamp with time zone)",
  ],
  "tables": Object {
    "test": Array [
      Object {
        "arrayDataType": null,
        "camelName": "id",
        "columnName": "id",
        "dataType": "integer",
      },
      Object {
        "arrayDataType": null,
        "camelName": "testName",
        "columnName": "test_name",
        "dataType": "text",
      },
    ],
    "version": Array [
      Object {
        "arrayDataType": null,
        "camelName": "version",
        "columnName": "version",
        "dataType": "numeric",
      },
      Object {
        "arrayDataType": null,
        "camelName": "deployStart",
        "columnName": "deploy_start",
        "dataType": "timestamp with time zone",
      },
      Object {
        "arrayDataType": null,
        "camelName": "deployedTs",
        "columnName": "deployed_ts",
        "dataType": "timestamp with time zone",
      },
      Object {
        "arrayDataType": null,
        "camelName": "clientCount",
        "columnName": "client_count",
        "dataType": "integer",
      },
      Object {
        "arrayDataType": null,
        "camelName": "payload",
        "columnName": "payload",
        "dataType": "jsonb",
      },
      Object {
        "arrayDataType": null,
        "camelName": "payloadVersion",
        "columnName": "payload_version",
        "dataType": "numeric",
      },
    ],
  },
}
`;

exports[`using pg to test migrations before migration the table and columns dont match 2`] = `
Object {
  "functions": Array [
    "js_time(timestamp with time zone)",
  ],
  "tables": Object {
    "test": Array [
      Object {
        "arrayDataType": null,
        "camelName": "id",
        "columnName": "id",
        "dataType": "integer",
      },
      Object {
        "arrayDataType": null,
        "camelName": "testName",
        "columnName": "test_name",
        "dataType": "text",
      },
      Object {
        "arrayDataType": "text",
        "camelName": "tags",
        "columnName": "tags",
        "dataType": "ARRAY",
      },
      Object {
        "arrayDataType": null,
        "camelName": "message",
        "columnName": "message",
        "dataType": "text",
      },
    ],
    "version": Array [
      Object {
        "arrayDataType": null,
        "camelName": "version",
        "columnName": "version",
        "dataType": "numeric",
      },
      Object {
        "arrayDataType": null,
        "camelName": "deployStart",
        "columnName": "deploy_start",
        "dataType": "timestamp with time zone",
      },
      Object {
        "arrayDataType": null,
        "camelName": "deployedTs",
        "columnName": "deployed_ts",
        "dataType": "timestamp with time zone",
      },
      Object {
        "arrayDataType": null,
        "camelName": "clientCount",
        "columnName": "client_count",
        "dataType": "integer",
      },
      Object {
        "arrayDataType": null,
        "camelName": "payload",
        "columnName": "payload",
        "dataType": "jsonb",
      },
      Object {
        "arrayDataType": null,
        "camelName": "payloadVersion",
        "columnName": "payload_version",
        "dataType": "numeric",
      },
    ],
  },
}
`;
